Detour/Source/DetourNavMesh.cpp

```diff
@@ -1302,6 +1302,9 @@ struct dtPolyState
 {
 	unsigned short flags;						// Flags (see dtPolyFlags).
 	unsigned char area;							// Area ID of the polygon.
+#ifdef NAVMESH_ISLAND_SYSTEM
+	unsigned int islandIdx;                     // Island index of the polygon
```

Comment:
When changing to just use an array for this data we should simply allow the island to be serialized separately.

Reply:
done